wallet-new/test/unit/UTxO/BlockGen.hs

```diff
@@ -219,3 +236,173 @@ zipFees
     :: [[Value -> Transaction h Addr]]
     -> ([[Value]] -> [[Transaction h Addr]])
 zipFees = zipWith (zipWith ($))
+
+{-------------------------------------------------------------------------------
+  Forkable blockchains
+-------------------------------------------------------------------------------}
+
+type NoFeeBlockTree h = Tree [Value -> Transaction h Addr]
+
+-- | Global context for building block trees.
+--   This structure is shared between all branches, and controls the
+--   extent to which the tree forks.
+data TreeGenGlobalCtx h = TreeGenGlobalCtx
+    { _treeGenGlobalCtxFreshHashSrc                :: !Int
+      -- ^ A fresh hash value for each new transaction.
+    {- The below values form part of configuration, and are unlikely to be
+       updated during tree building.
+    -}
+    , _treeGenGlobalCtxInputPartiesUpperLimit      :: !Int
+      -- ^ The upper limit on the number of parties that may be selected as
+      -- inputs to a transaction.
+    , _treeGenGlobalCtxForkLikelihood              :: !Double
+      -- ^ Likelihood of generating a forked block at any point in the tree.
+      --   Only the non-integer part of this value is considered.
+    , _treeGenGlobalCtxPruneLikelihood             :: !Double
+      -- ^ Likelihood of the non-principal branch to be terminated at any
+      --   block.
+    , _treeGenGlobalCtxMaxHeight                   :: !Int
```

Comment:
What's the policy regarding machine-sized types in situations like this? They make an otherwise readily serializable structure less so. Moreover, shouldn't unsigned `Number` or `Word*` types be preferred in general?

Reply:
Agree on this point